### PR TITLE
Remove external filesystem

### PR DESCRIPTION
**Pull request by phlptp**
### Summary

If merged this pull request will update minimum compiler and CMake requirements and use std::filesystem

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,7 +8,21 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
...
+
+### Fixed
+
+### Changed
+
+### Added
+
+### Removed
+
+- included filesystem header files, since all minimum compilers have that header available.
```
Comment (nightlark) [inline CHANGELOG.md:23]:
```suggestion
- ghc::filesystem, since all minimum compilers have support for std::filesystem available.
```

**Comment by phlptp** [comment]
@nightlark I think this is ready for review now

File: src/helics/network/NetworkCommsInterface.cpp [hunk 2]
```diff
@@ -17,23 +17,29 @@ SPDX-License-Identifier: BSD-3-Clause
...
-static const std::string localHostString = "localhost";
+NetworkCommsInterface::PortAllocator::PortAllocator()
+{
+    addNewHost(localHostString);
+}
 
-int NetworkCommsInterface::PortAllocator::findOpenPort(int count, const std::string& host)
+int NetworkCommsInterface::PortAllocator::findOpenPort(int count, std::string_view host)
 {
     if ((host == "127.0.0.1") || (host == "::1")) {
```
[the remark this answers is not in the document]
Comment (phlptp) [reply src/helics/network/NetworkCommsInterface.cpp:36]: maybe, I might be converting from localhost to 127.0.0.1 earlier in the code stack though so would have to did to get a definite answer

**Review by nightlark — APPROVED** [review]

File: src/helics/network/NetworkCommsInterface.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (nightlark) [reply src/helics/network/NetworkCommsInterface.cpp:36]: Actually, it looks like the function expects "localhost" and if it sees  "127.0.0.1" or "::1" then it finds open ports using "localhost".

**Pull request merged**